File: backend/MedTrack.API/MedTrack.API/Models/MedicalInfo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MedTrack.API.Models
{
    public class MedicalInfo
    {
        [Key]
        public int MedicalInfoId { get; set; }

        [ForeignKey("Patient")]
        public int UserId { get; set; }

        public string Allergies { get; set; }
        public string Medications { get; set; }
        public bool Smoking { get; set; }
        public bool Alcohol { get; set; }
        public string PhysicalActivity { get; set; }

        public Patient? Patient { get; set; }
    }
}

File: frontend/medtrack-frontend/src/api/medicalInfo.js
import axios from 'axios'

/**
 * Fetch a patient’s medical info
 * GET /api/MedicalInfo/{patientId}
 */
export function getMedicalInfo(patientId) {
  return axios
    .get(`/api/MedicalInfo/${patientId}`)
    .then(res => res.data)
}

/**
 * Create a new medical info record
 * POST /api/MedicalInfo
 * payload = { patientId, allergies, medications, smoking, alcohol, physicalActivity }
 */
export function createMedicalInfo(payload) {
  return axios
    .post('/api/MedicalInfo', payload)
    .then(res => res.data)
}

/**
 * Update an existing medical info record
 * PUT /api/MedicalInfo/{patientId}
 * payload = { allergies, medications, smoking, alcohol, physicalActivity }
 */
export function updateMedicalInfo(patientId, payload) {
  return axios
    .put(`/api/MedicalInfo/${patientId}`, payload)
    .then(res => res.data)
}

/**
 * Delete a patient’s medical info record
 * DELETE /api/MedicalInfo/{patientId}
 */
export function deleteMedicalInfo(patientId) {
  return axios
    .delete(`/api/MedicalInfo/${patientId}`)
    .then(res => res.data)
}
